packages/material-ui/src/Slider/Slider.d.ts

```diff
@@ -209,4 +245,30 @@ export type SliderProps<
   P = {}
 > = OverrideProps<SliderTypeMap<P, D>, D>;
 
+type SliderRootProps = Omit<SliderProps, 'components' | 'componetnsProps'>;
```

Comment:
```suggestion
type SliderRootProps = SliderProps["componentsProps"]["root"]
```

I tried doing this but the ts compiler is complaining because root is optional 😢 I duplicated them for now, but if someone has a better suggestion please let me know. FYI @oliviertassinari

Reply:
There is a workaround using `NonNullable` https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype

```ts
type SliderRootProps = NonNullable<SliderProps["componentsProps"]>["root"]
```

https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgZQBsEATNABSgjAGc4BvAKDhbmwgFtIlgUbLqNAPwAuBs1aSqEGKIYBfCS0WLGMAJ5g8RUmgBKEGQNoBGOAF4CxMlGM0A2gCJ2XCDz53HAXSfSY3uDVNbWt9Qxg7ACYLOAA5N1iAV0JCTAAjQmAAHh0bOycXbl4YfipabwA+X3DvRiA

Not sure if it's better in any way.

____

Maybe it's better to go another way and define child component props first, and then use it the main component. But it will require A LOT of refactorings, and code diff will be a mess. 

```ts
type SliderRootProps = { … };
type SliderMarkLabelProps = { … };
… 
export interface SliderTypeMap<P = {}, D extends React.ElementType = 'span'> {
  props: P & {
    componentProps?: {
      root?: SliderRootProps,
      markLabel?: SliderMarkLabelProps,
      …
    }
  }
}
```

Reply:
`SliderRootProps` it's basically all SliderProps, so we would have circular reference. But I will try the `NonNullable`, wasn't aware of that, thanks for recommending it 🙇